File: src/main/kotlin/utils/Formatters.kt
package utils

import core.TeamComposition

fun TeamComposition.format(): String {
    val skills = composition.joinToString("\n\n") { role ->
        val (skill, member) = role

        """
        |Skill:  ${skill.language}
        |Name:   ${member.name.value}
        |Avatar: ${member.avatarUrl.value}
        |GitHub: ${member.profileUrl.value}
        |Skills: ${member.skills.joinToString()}
        """.trimMargin()
    }

    return """
    |Team Composition
    |
    |Skills
    |
    |$skills
    """.trimMargin()
}

File: frontend/lib/utils/formatters.dart
import 'package:code_connect_common/code_connect_common.dart';

extension TeamCompositionDescription on TeamComposition {
  String describe() {
    final skills = roles.map((role) => role.describe()).join('\n\n');

    return '''
Team Composition:

[Description]
$projectDescription

[Roles]
$skills

'''
        .trimLeft();
  }
}

extension ProjectRoleDescription on ProjectRole {
  String describe() {
    return '''
Skill:  ${skill.language}
Name:   ${member.name}
Avatar: ${member.avatarUrl}
GitHub: ${member.profileUrl}
Stack:  ${member.skills.join()}
'''
        .trimLeft();
  }
}
